core/task-impl.h

```diff
@@ -0,0 +1,13 @@
+#ifndef BESS_TASK_IMPL_H_
+#define BESS_TASK_IMPL_H_
+
+#include "task.h"
+
+#include "module.h"
+
+inline struct task_result Task::operator()(void)
```

Comment:
Good point.  If I forward declare `Task` in `worker.h`, I can get rid of `task.h` and `task-impl.h`